### PR TITLE
1.7.7, 1.7.8 Cherry picks

### LINKED ISSUE #16361: 3.7 pick list
this is a list of items in the release-1.7 branch we need to pick or rebase to pick up. Given the number of items, a rebase onto 1.7.x seems the most reasonable approach.

TBD (mostly in 1.7.8, some we already had picked):
- [ ] `#52933: Preserve leading and trailing slashes on proxy subpaths`
- [ ] `#52176: Use separate client for node status loop`
- [ ] `#52545: use specified discovery information if possible`
- [ ] `#43152: etcd3 store: retry w/live object on conflict`
- [ ] `#48394: GuaranteedUpdate must write if stored data is not canonical`
- [ ] `#53239: Correct APIGroup for RoleBindingBuilder Subjects`

1.7.6:
- [x] `#52385:  Add --request-timeout to allow the global request timeout of 60 seconds to be configured.`
- [x] `#52329: Use credentials from providers for docker sandbox image`
- [x] `#52228: Fix discovery restmapper finding resources in non-preferred versions`
- [x] `#52103: Fix proxied request-uri to be valid HTTP requests`
- [x] `#52092: Fix resource quota controller panic`
- [x] `#51969: Fixes an issue with APIService auto-registration affecting rolling HA apiserver restarts that add or remove API groups being served.`
- [x] `#51954: Fix providerID update validation`
- [x] `#51871: Use local JSON log buffer in parseDockerJSONLog.`
- [x] `#51763: change SizeLimit to a pointer`

1.7.5:
- [x] `#51177: On AttachDetachController node status update, do not retry when node doesn't exist but keep the node entry in cache`
- [x] `#51469: Enable finalizers independent of GC enablement`
- [x] `#50862: Fix premature return in strategicmergepatch`
- [x] `#50596: Allow unmounting bind-mounted directories.`
- [x] `#51149: StatefulSet controller no longer attempts to mutate v1.PodSpec.Hostname or v1.PodSpec.Subdomain`
- [x] `#51127: Fix initial exec terminal dimensions`
- [x] `#49973: ScaleIO Volume Plugin - volume attribute updates`
- [x] `#49767: make it possible to allow discovery errors for controllers`

1.7.4:
- [x] `#50623: csr: add resync to csr approver`
- [x] `#50260: AWS:  We never want to modify the globally defined SG`
- [x] `#50513: AttachDisk should not call detach inside of Cinder volume provider`
- [x] `#50250: fix data race in storage (during addition)`
- [x] `#50240: Fix incorrect call to 'bind' in scheduler, Retry scheduling pods after errors more consistently in scheduler`
- [x] `#50030: Make NodeRestriction admission allow evictions for bounded pods`
- [x] `#50015: Correctly handle empty watch event cache`

1.7.3:
- [x] `#49768: fix pdb validation bug on spec` ([PR](https://github.com/kubernetes/kubernetes/pull/49768))
- [x] `#49781: fix check for nil value in interface for proxier health` ([PR](https://github.com/kubernetes/kubernetes/pull/49781))
- [x] `#49582: Fix issues for local storage allocatable feature` ([PR](https://github.com/kubernetes/kubernetes/pull/49582))
- [x] `#49578: Fix issue when setting fileysystem capacity in container` ([PR](https://github.com/kubernetes/kubernetes/pull/49578))
- [x] `#49592: Fix Operation names for subresources` ([PR](https://github.com/kubernetes/kubernetes/pull/49592))
- [x] `#49403: Use specified ServerName in aggregator TLS validation` ([PR](https://github.com/kubernetes/kubernetes/pull/49403))
- [x] `#49447: Do not spin forever if kubectl drain races with other removal` ([PR](https://github.com/kubernetes/kubernetes/pull/49447))
- [x] `#49455: Fix controller crash because of nil volume spec` ([PR](https://github.com/kubernetes/kubernetes/pull/49455))
- [x] `#49026: rebase gophercloud to support HTTP status 300 in pagination, so listingng Cinder v1/v2 API versions won't break` ([PR](https://github.com/kubernetes/kubernetes/pull/49026))
- [x] `#49254: Restore cAdvisor prometheus metrics to the main port` ([PR](https://github.com/kubernetes/kubernetes/pull/49254))
- [x] `#49381: glusterfs: retry without auto_unmount only when it's not supported` ([PR](https://github.com/kubernetes/kubernetes/pull/49381))

1.7.2:
- [x] `#49155: make externalAdmissionHookConfigurationManager distinguish API disabled error` ([PR](https://github.com/kubernetes/kubernetes/pull/49155))
- [x] `#49067: Use local PX endpoint for mount, unmount, detach and attach calls` ([PR](https://github.com/kubernetes/kubernetes/pull/49067))
- [x] `#49089: Fix subPath existence check to not follow symlink` ([PR](https://github.com/kubernetes/kubernetes/pull/49089))
- [x] `#49069: Never prevent deletion of resources as part of namespace` ([PR](https://github.com/kubernetes/kubernetes/pull/49069))
- [x] `#48821: Correctly filter terminated pods in kubectl` ([PR](https://github.com/kubernetes/kubernetes/pull/48821))
- [x] `#48809: fix udp service blackhole problem when number of backends changes from 0 to non-0, local storage teardown fix` ([PR](https://github.com/kubernetes/kubernetes/pull/48809))

1.7.1:
- [x] `#49047: hpa: Prevent scaling below MinReplicas if desiredReplicas is zero` ([PR](https://github.com/kubernetes/kubernetes/pull/49047))
- [x] `#48855: GCE: Fix panic when service loadbalancer has static IP address` ([PR](https://github.com/kubernetes/kubernetes/pull/48855))
- [x] `#48542: Ensure namespace exists as part of RBAC reconciliation` ([PR](https://github.com/kubernetes/kubernetes/pull/48542))
- [x] `#48490: Support NoSchedule taints correctly in DaemonSet controller` ([PR](https://github.com/kubernetes/kubernetes/pull/48490))
- [x] `#48261: Fix removing finalizer for garbage collector` ([PR](https://github.com/kubernetes/kubernetes/pull/48261))

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #16361

### CODE REVIEW

**Comment by soltysh** [comment]
@ironcladlou fyi

**Comment by liggitt** [comment]
picks look fine, tests need looking into.

unit test is legitimate:
```
In suite "github.com/openshift/origin/pkg/cmd/server/bootstrappolicy", test case "TestGetBootstrapNamespaceRoleBindings" failed:
=== RUN   TestGetBootstrapNamespaceRoleBindings
--- FAIL: TestGetBootstrapNamespaceRoleBindings (0.02s)
	policy_test.go:124: Bootstrap policy data does not match the test fixture in ../../../../test/testdata/bootstrappolicy/bootstrap_namespace_role_bindings.yaml
	policy_test.go:135: Diff between bootstrap data and fixture data in ../../../../test/testdata/bootstrappolicy/bootstrap_namespace_role_bindings.yaml:
```

couln't actually spot the failed integration test

**Comment by soltysh** [comment]
Added missing import, I'll apply the labels back when tests will be green.

File: pkg/build/registry/buildconfig/webhook.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (h *WebHookHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // ProcessWebHook does the actual work of processing the webhook request
 func (w *WebHookHandler) ProcessWebHook(writer http.ResponseWriter, req *http.Request, ctx apirequest.Context, name, subpath string) error {
-	parts := strings.Split(subpath, "/")
+	parts := strings.Split(strings.TrimPrefix(subpath, "/"), "/")
```
Comment (soltysh) [inline pkg/build/registry/buildconfig/webhook.go]: @bparees fyi

Comment (bparees) [reply pkg/build/registry/buildconfig/webhook.go]: @soltysh thanks....  k8s changed something about how the provide the request path to us??

Comment (liggitt) [reply pkg/build/registry/buildconfig/webhook.go]: Subresource path was stripping leading/trailing slashes incorrectly. Fixed in 1.7.8

Comment (soltysh) [reply pkg/build/registry/buildconfig/webhook.go]: Yup, just that.

**Comment by soltysh** [comment]
Applying the lgtm label on green tests.

**Pull request merged**